### PR TITLE
Redis expire

### PR DESCRIPTION
**Pull request by CCharlieLi**
- Add connect test to redis server
- Deploy create(with EXPIRE) test

### CODE REVIEW

**Comment by makara** [comment]
Thanks. Please add a test:

- Set TTL to 3
- Save a model
- Wait 3 (or more) seconds and find the model and it should be gone.

File: backends/redis.js [hunk 1]
```diff
@@ -5,9 +5,12 @@ module.exports = function(Model, options) {
   if (options.ttl != null && options.ttl) {
     // Assuming the Model connector is "redis" here but we may need a config in
     // the future.
+
     Model.observe('after save', function(ctx, next) {
       // @see http://redis.io/commands/expire.
-      // TODO
+      var redisClient = ctx.Model.getConnector().client._client;
```
Comment (makara) [inline backends/redis.js]: I believe `client` has proxy to all the `_client` methods. See https://github.com/strongloop/loopback-connector-redis/blob/master/lib/redis.js#L71. Please use that if possible.

**Comment by CCharlieLi** [comment]
Thanks for your reminding.
- I've updated the test, the saved item will be expired in 3s. The test should pass, If not it could be becouse of the latency.
- Update to use connector's redis client.

File: test/CRUD.test.js [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+      mixin(Person,config);
+
+      Person.create({
+        id: 3,
+        name: 'Charlie',
+        age: 24
+      }, function(err, res) {
+        if(err) console.log(err);
+        //block
+        for(var start = +new Date; +new Date - start <= config.ttl*1000; ) { }
```
Comment (makara) [inline test/CRUD.test.js]: `setTimeout(`)

File: backends/redis.js [hunk 3]
```diff
@@ -5,9 +5,13 @@ module.exports = function(Model, options) {
   if (options.ttl != null && options.ttl) {
     // Assuming the Model connector is "redis" here but we may need a config in
     // the future.
+
     Model.observe('after save', function(ctx, next) {
       // @see http://redis.io/commands/expire.
-      // TODO
+      var redisClient = ctx.Model.getConnector().client;
+      //pass arguments as array
+      redisClient.expire([ctx.Model.modelName+":"+ctx.instance.id, options.ttl]);
```
Comment (makara) [inline backends/redis.js]: It'd expect a callback, you could pass the `next` to it but let me know if you got different opinions.

Comment (CCharlieLi) [inline backends/redis.js]: How about pass a callback function to throw a Error? Cause we may deploy other methods in 'after save'.

**Comment by CCharlieLi** [comment]
@makara   As we may deploy other methods in 'after save' section, I'm not sure if pass `next` to `expire()` is good, how about pass a callback to output the error?

**Comment by makara** [comment]
No, the errors are always need to be handled. If we don't know how to handle them, leave it to the users. We can, however, provide a way to the users to distinguish it from other errors like for example, build a new error class or something.

**Pull request merged**